napari/layers/_points_layer/model.py

```diff
@@ -103,26 +167,40 @@ def coords(self) -> np.ndarray:
 
     @coords.setter
     def coords(self, coords: np.ndarray):
+        cur_npoints = len(self._coords)
         self._coords = coords
 
         # Adjust the size array when the number of points has changed
-        if len(coords) < len(self._size):
+        if len(coords) < cur_npoints:
             # If there are now less points, remove the sizes of the missing
             # ones
             with self.freeze_refresh():
-                self.size = self._size[: len(coords)]
-        elif len(coords) > len(self._size):
+                self.size_array = self._size_array[: len(coords)]
+                self._edge_color_list = self._edge_color_list[: len(coords)]
+                self._face_color_list = self._face_color_list[: len(coords)]
+
+        elif len(coords) > cur_npoints:
```

Comment:
no this will then pad the `size_array`, `_edge_color_list`, and `_face_color_list` so that they go from length 10 to length 12 using the new sizes / colors. I've tried to improve the comments. Let me know if it is still unclear or if there is something else confusing